### DISCUSSION TITLE
Enable TCP keep alive option

### DISCUSSION THREAD

**Discussion by macsir** [cubewise-code/tm1py#804] [Ideas]
Hi @MariusWirtz Have you had any experience of finding some timeout when trying to receive a response from TM1py execute with return function while running a long-run TI in any cloud environment? I might submit a PR to let user choose to use this socket option and it would be helpful when encountering this similar situation. Thanks

**Comment by MariusWirtz** [comment]
To work around a timeout you can use the `async_requests_mode`.
It was initially implemented to work with the fixed 60s timeouts in the IBM cloud environments but it works with any TM1 Server.

However, if you find a more efficient solution on the socket level I would appreciate a PR very much!

**Comment by macsir** [comment]
Hi, @MariusWirtz Thanks for the suggestion. I see you points and it works in my AWS environment. As you said, it keeps polling every second and not very efficient. Also, I have tested tcp keepalive option in the TM1py 1.9 and works well also without a bit extra waiting time. I would submit that PR soon after more test cases.
